main.go

```diff
@@ -34,12 +39,28 @@ func main() {
 	}
 
 	var db *gorm.DB
+	var sqlDb *sql.DB
 	if strings.HasPrefix(cfg.DatabaseUri, "postgres://") || strings.HasPrefix(cfg.DatabaseUri, "postgresql://") || strings.HasPrefix(cfg.DatabaseUri, "unix://") {
-		db, err = gorm.Open(postgres.Open(cfg.DatabaseUri), &gorm.Config{})
-		if err != nil {
-			log.Fatalf("Failed to open DB %v", err)
+		if os.Getenv("DATADOG_AGENT_URL") != "" {
+			sqltrace.Register("pgx", &stdlib.Driver{}, sqltrace.WithServiceName("nostr-wallet-connect"))
+			sqlDb, err = sqltrace.Open("pgx", cfg.DatabaseUri)
+			if err != nil {
+				log.Fatalf("Failed to open DB %v", err)
+			}
+			db, err = gormtrace.Open(postgres.New(postgres.Config{Conn: sqlDb}), &gorm.Config{})
```

Comment:
Same comment as in the other PR's, I think you can use the same syntax in the Open arguments?